agent/engine/docker_task_engine.go

```diff
@@ -303,6 +308,14 @@ func (engine *DockerTaskEngine) sweepTask(task *api.Task) {
 			seelog.Errorf("Error removing container reference from image state: %v", err)
 		}
 	}
+
+	// Clean metadata directory for task
+	if engine.cfg.ContainerMetadataEnabled {
```

Comment:
This check seems weak, as writing the metadata file could fail. The safer way is to add a new state of this file and only delete this file if it's created.

Also please add a debug log here to indicate this happened.